cmd/bosun/sched/host.go

```diff
@@ -0,0 +1,809 @@
+				TimedOutLastUpdated:    timestamp,
+				DNSResolved:            dnsLookup == 1 && dnsErr == nil,
+				DNSResolvedLastUpdated: dnsTimestamp,
+				RTTMS:          rtt,
+				RTTLastUpdated: rttTimestamp,
+			}
+
+		}
+		// Process Hardware Chassis States
+		for _, ts := range hwChassisTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			component, ok := ts["component"]
+			if !ok {
+				return nil, fmt.Errorf("couldn't find component tag for host %s", host.Name)
+			}
+			fstatus, timestamp, err := s.Search.GetLast("hw.chassis", ts.String(), false)
+			status := "Bad"
+			if fstatus == 0 {
+				status = "Ok"
+			}
+			if err == nil && timestamp > 0 {
+				host.Hardware.ChassisComponents[component] = &ChassisComponent{
+					Status:            status,
+					StatusLastUpdated: timestamp,
+				}
+			}
+		}
+		for _, ts := range hwTempsTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			name, ok := ts["name"]
+			if !ok {
+				slog.Errorf("couldn't find name tag %s for host %s", host.Name, name)
+			}
+			tStatus, timestamp, err := s.Search.GetLast("hw.chassis.temps", ts.String(), false)
+			celsius, rTimestamp, err := s.Search.GetLast("hw.chassis.temps.reading", ts.String(), false)
+			status := "Bad"
+			if tStatus == 0 {
+				status = "Ok"
+			}
+			if err == nil && timestamp > 0 {
+				host.Hardware.Temps[name] = &Temp{
+					Celsius:            celsius,
+					Status:             status,
+					StatusLastUpdated:  timestamp,
+					CelsiusLastUpdated: rTimestamp,
+				}
+			}
+		}
+		for _, ts := range hwPowerSuppliesTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			id, ok := ts["id"]
+			if !ok {
+				return nil, fmt.Errorf("couldn't find power supply tag for host %s", host.Name)
+			}
+			idPlus, err := strconv.Atoi(id)
+			if err != nil {
+				slog.Errorf("couldn't conver it do integer for power supply id %s", id)
+			}
+			idPlus++
+			fstatus, timestamp, err := s.Search.GetLast("hw.ps", ts.String(), false)
+			status := "Bad"
+			if fstatus == 0 {
+				status = "Ok"
+			}
+			current, currentTimestamp, _ := s.Search.GetLast("hw.chassis.current.reading", opentsdb.TagSet{"host": host.Name, "id": fmt.Sprintf("PS%v", idPlus)}.String(), false)
+			volts, voltsTimestamp, _ := s.Search.GetLast("hw.chassis.volts.reading", opentsdb.TagSet{"host": host.Name, "name": fmt.Sprintf("PS%v_Voltage_%v", idPlus, idPlus)}.String(), false)
+			ps := &PowerSupply{}
+			if err == nil && timestamp > 0 {
+				ps.Status = status
+				ps.StatusLastUpdated = timestamp
+				ps.Amps = current
+				ps.AmpsLastUpdated = currentTimestamp
+				ps.Volts = volts
+				ps.VoltsLastUpdated = voltsTimestamp
+				host.Hardware.PowerSupplies[id] = ps
+			}
+			for _, m := range hostMetadata {
+				if m.Time.Before(time.Now().Add(-timeFilterAge)) {
+					continue
+				}
+				if !m.Tags.Equal(ts) {
+					continue
+				}
+				switch val := m.Value.(type) {
+				case string:
+					switch m.Name {
+					case "psMeta":
```

Comment:
maybe check for name = psMeta earlier in the metadata loop. It should be cheaper than tagset equals.